posts/tests.py

```diff
@@ -155,3 +155,44 @@ def test_login_wrong_password(self):
     def test_unregistered_user_login(self):
         response = self.client.post(reverse('login'), self.login3, follow=True)
         self.assertFalse(response.context['user'].is_active)
+
+class LoggedInUserArticleCreationTest(TestCase):
+
+    def setUp(self):
+        self.signup1 = {'username': 'testuser2', 'password1': 'aaaa1234', 'password2': 'aaaa1234'}
+        self.login = {'username': 'testuser2', 'password': 'aaaa1234'}
+        User.objects.create_user(**self.login)
+        self.input = {'title':'What a super day', 'body':'this would have been a great day if...'}
+
+    def test_login_post_creation(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
```

Comment:
Post.objects.get() will raise `MultipleObjectsReturned` error when there is more than a single Post.
Try to extract  id from the redirect url and use Post.object.get(id='')

Reply:
And then validate fields of instance with input data